.github/workflows/main.yml

```diff
@@ -1,15 +1,21 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```

Comment:
I no longer remember why I did this.

Reply:
Whenever someone has a PR open on a branch, it  runs the whole CI _twice_ every time they push a commit--once because they pushed a commit, and once because its a PR. 

I believe this would fix that. It should probably be a separate PR though.